WinTLS Error

I frequently receive the following error message: 
`[ERROR] WinTLS: Failed to encrypt a message! Error: The specified data could not be encrypted. (80090329)`
The download seems to be unaffected and completes successfully, as far as I can tell.

A _similar_ error is also mentioned here: https://github.com/aria2/aria2/issues/351
That thread suggests the cause is WinXP, however I am not using XP.

What does this error mean?

--- a/src/WinTLSSession.cc
+++ b/src/WinTLSSession.cc
@@ -386,8 +386,16 @@ ssize_t WinTLSSession::writeData(const void* data, size_t len)
                 static_cast<char*>(sendRecordBuffers_[1].pvBuffer));
     status_ = ::EncryptMessage(&handle_, 0, &desc, 0);
     if (status_ != SEC_E_OK) {
-      A2_LOG_ERROR(fmt("WinTLS: Failed to encrypt a message! %s",
-                       getLastErrorString().c_str()));
+      if (state_ != st_closing) {
+        A2_LOG_ERROR(fmt("WinTLS: Failed to encrypt a message! %s",
+                         getLastErrorString().c_str()));
+      } else {
+        // On closing state, don't log this message in error level
+        // because it seems that the encryption tends to fail in that
+        // state.
+        A2_LOG_DEBUG(fmt("WinTLS: Failed to encrypt a message! %s",
+                         getLastErrorString().c_str()));
+      }
       state_ = st_error;
       return TLS_ERR_ERROR;
     }

